### PR TITLE
Feature/issue191 wipp library refactor angular upgrade

### PR DESCRIPTION
**Pull request by mohamedOuladi**
## What does this PR do?

- Refactored the routing in the library: instead of using hard coded paths in components, the paths are now injected into the library from the env file
- Upgraded angular version from 7 to 9: upgraded other 3rd party libraries as well to work properly with angular 9
- Added Generic Data upload feature

## Related issues

#191 https://github.com/usnistgov/WIPP-frontend/issues/191
#192 https://github.com/usnistgov/WIPP-frontend/issues/192
#184 https://github.com/usnistgov/WIPP-frontend/issues/184

### CODE REVIEW

**Comment by mohamedOuladi** [comment]
FYI: I disabled inline editor library because it's causing issues with angular 9. We might need to look into an alternative solution to give the user the ability to modify collections' names.

File: deploy/docker/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-3.1.0
+3.1.3
```
Comment (sunnielyu) [inline deploy/docker/VERSION]: I would bump the minor version. The patch version should be reserved for bug fixes.

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts [hunk 2]
```diff
@@ -1,34 +1,117 @@
...
+import {MatPaginator} from '@angular/material/paginator';
 
 @Component({
   selector: 'app-generic-data-detail',
   templateUrl: './generic-data-detail.component.html',
   styleUrls: ['./generic-data-detail.component.css']
 })
-export class GenericDataDetailComponent implements OnInit {
+
+// @NgModule({
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: This module block can be removed

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts [hunk 3]
```diff
@@ -1,34 +1,117 @@
-import { Component, OnInit } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, NgModule, OnInit, ViewChild } from '@angular/core';
 import {Job} from '../../job/job';
-import {ActivatedRoute} from '@angular/router';
-import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import {ActivatedRoute, Router} from '@angular/router';
+import {NgbModal, NgbModule} from '@ng-bootstrap/ng-bootstrap';
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: NgbModule import can be removed

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts [hunk 4]
```diff
@@ -1,34 +1,117 @@
-import { Component, OnInit } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, NgModule, OnInit, ViewChild } from '@angular/core';
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: NgModule import can be removed

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts [hunk 5]
```diff
@@ -1,34 +1,117 @@
...
+import {ActivatedRoute, Router} from '@angular/router';
+import {NgbModal, NgbModule} from '@ng-bootstrap/ng-bootstrap';
 import {JobDetailComponent} from '../../job/job-detail/job-detail.component';
 import {GenericData} from '../generic-data';
+import {GenericFile} from '../generic-file';
 import {GenericDataService} from '../generic-data.service';
+import {BehaviorSubject, Observable, of as observableOf, Subject} from 'rxjs';
+import * as Flow from '@flowjs/flow.js';
+import {auditTime, catchError, map, switchMap} from 'rxjs/operators';
+import {BytesPipe, NgMathPipesModule} from 'angular-pipes';
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: NgMathPipesModule can be removed

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts [hunk 6]
```diff
@@ -42,4 +125,82 @@ export class GenericDataDetailComponent implements OnInit {
...
+    this.flowHolder.assignBrowse([this.browseDirBtn.nativeElement], true, false);
+    this.flowHolder.assignDrop(this.dropArea.nativeElement);
+
+    const id = '';
+    const genericFileUploadUrl = this.genericDataService.getGenericFileUrl(this.genericData);
+    this.flowHolder.opts.target = genericFileUploadUrl;
+
+    const self = this;
+    this.flowHolder.on('fileAdded', function (file, event) {
+      console.log(file, event);
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: Remove console.log

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-new/generic-data-new.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+              private modalService: NgbModal,
+              private genericDataService: GenericDataService,
+              private router: Router) {
+  }
+
+  ngOnInit() {
+  }
+
+  createCollection() {
+    console.log('createCollection');
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/generic-data/generic-data-new/generic-data-new.component.ts]: Remove console.log

File: projects/wipp-frontend-lib/src/lib/stitching-vector/stitching-vector-detail/stitching-vector-detail.component.ts [hunk 8]
```diff
@@ -39,6 +39,9 @@ export class StitchingVectorDetailComponent implements OnInit {
         this.stitchingVector = stitchingVector;
         this.getJob();
       });
+
+    console.log('hello');
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/stitching-vector/stitching-vector-detail/stitching-vector-detail.component.ts]: Remove console.log

File: projects/wipp-frontend-lib/src/lib/dynamic-content/dynamic-content.component.ts [hunk 9]
```diff
@@ -5,12 +5,33 @@ import {
...
+import { TensorboardLogsTemplateComponent } from '../tensorflow-model/tensorflow-model-template/tensorboard-logs-template.component';
+
+
+export const entryComponentsMap = {
+  'collection': ImagesCollectionTemplateComponent,
+  'stitchingVector': StitchingVectorTemplateComponent,
+  'csvCollection': CsvCollectionTemplateComponent,
+  'pyramid': PyramidTemplateComponent,
+  'pyramidAnnotation': PyramidAnnotationTemplateComponent,
+  'notabook': NotebookTemplateComponent,
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/dynamic-content/dynamic-content.component.ts]: typo

File: projects/wipp-frontend-lib/src/lib/dynamic-content/dynamic-content.component.ts [hunk 10]
```diff
@@ -43,13 +64,17 @@ export class DynamicContentComponent implements OnInit, OnDestroy {
   constructor(private componentFactoryResolver: ComponentFactoryResolver) {}
 
   ngOnInit() {
+    var factoryClass;
```
Comment (sunnielyu) [inline projects/wipp-frontend-lib/src/lib/dynamic-content/dynamic-content.component.ts]: Use `let` instead of `var

**Review by sunnielyu — CHANGES_REQUESTED** [review]

File: deploy/docker/VERSION (hunk 1 above)
Comment (mohamedOuladi) [reply deploy/docker/VERSION]: fixed

File: projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts (hunk 2 above)
Comment (mohamedOuladi) [reply projects/wipp-frontend-lib/src/lib/generic-data/generic-data-detail/generic-data-detail.component.ts]: fixed

**Review by sunnielyu — APPROVED** [review]

**Pull request merged**